core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java

```diff
@@ -23,7 +22,6 @@ public interface ConfigConfig {
     /**
      * Profile that will be active when Quarkus launches.
      */
-    @WithDefault("prod")
```

Comment:
I can see how we wouldn't want a default for an Optional but do you confirm this doesn't change our behavior?

Reply:
Yes, it is fine because the profile was always being overridden by the launch mode:
https://github.com/quarkusio/quarkus/pull/35322/files#diff-11168005e2c503c75c2ff0f927c06e4ca8fb9676b91d58e37ea98b05d68cc27dR29

So, there's no point in setting it here. Also, it would show up as the default in the docs, and because the profile changes depending on the launch mode, it doesn't make sense either to show a default profile in there.

Reply:
Thanks for confirming! Let's get this in then.